controller/characterController.go

```diff
@@ -0,0 +1,83 @@
+package controller
+
+import (
+	"os"
+	"encoding/csv"
+	"strconv"
+	"io"
+	
+	"github.com/carRub/academy-go-q12021/model"
+	"github.com/carRub/academy-go-q12021/usecase"
+)
+
+// CharacterController defines what a controller must contain
+type CharacterController struct {
+	characterInteractor usecase.CharacterInteractor
+}
+
+// CharacterUseCase defines functions to refer to our use cases
+type CharacterUseCase interface {
+	GetCharacters() ([]model.Character)
+	GetCharacterByID(id int) (*model.Character) 
+}
+
+// NewCharacterController Creates a new character controller 
+func NewCharacterController (ci usecase.CharacterInteractor) CharacterController {
+	return CharacterController{ci}
+}
+
+func (c CharacterController) GetCharacters() ([]model.Character, error) {
+	csvFile, _ := os.Open("assets/characters.csv")
```

Comment:
Never omit errors.

The route should not be hardcoded in the function (use func parameters to obtain it, or the package struct). Also, opening the file is not a task of a controller; you should move this logic into another package.

Reply:
This applies to the GetCharacterByID() method.